### PR TITLE
[JENKINS-62175] Restyle side panel widgets

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-62175](https://issues.jenkins-ci.org/browse/JENKINS-62175).

This PR revamps the styles for the side panel widgets, based on the designs shared on the UX SIG. 
I have focused on modifying as little markup within the widgets as possible (for example, I didn't even touch the HTML for the build history widget).

Unfortunately I wasn't able to test widgets provided by some plugins, such as cloud stats or next executions. I welcome any help or pointers to test them.

<details>
<summary>Screenshots</summary>

Idle queue & executors
Captura de pantalla 2020-06-02 a las 12 23 08

Busy queue & running executors
Captura_de_pantalla_2020-06-02_a_las_12_08_23

History 1
Captura_de_pantalla_2020-06-02_a_las_12_08_39

History 2
Captura_de_pantalla_2020-06-02_a_las_12_08_51

</details>

### Proposed changelog entries

* Entry 1: Side panel widgets restyled to have a more modern look & feel.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@timja 
@uhafner 
@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by alecharp** [comment]
The build failure seems to be infra related. Closing to retrigger a build.

**Comment by daniel-beck** [comment]
Merge conflict needs resolution.

**Comment by fqueiruga** [comment]
merge conflicts are fixed

File: war/src/main/less/modules/side-panel-widgets.less [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    max-width: 296px;
+  }
+}
+.build-row .build-name-controls .pane.build-name,
+.build-row .build-details-controls .pane.build-details {
+  width: 70%;
+}
+.build-row .build-row-cell .pane,
+#side-panel .build-row .build-row-cell .pane,
+// .build-row .build-row-cell .pane:first-child,
```
Comment (varyvol) [inline war/src/main/less/modules/side-panel-widgets.less:174]: I guess these can be removed now?

File: war/src/main/less/modules/side-panel-widgets.less [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  left: 14px;
+  border-left: 1px solid var(--panel-border-color--hover);
+}
+.build-row-cell .pane.build-name .display-name {
+  margin-left: 20px;
+}
+.build-row-cell .indent-multiline {
+  padding-left: 20px !important;  /* Sync changes with func expandControlsTo50Percent in hudson-behavior.js */
+}
+.build-row .build-row-cell {
```
Comment (varyvol) [inline war/src/main/less/modules/side-panel-widgets.less:201]: Ditto.

**Comment by oleg-nenashev** [comment]
AFAICT this increases the vertical size, right? I am not too concerned about it, just woder what would be the UX with a long queue

**Comment by fqueiruga** [comment]
It does increase the vertical size a little bit indeed, but causes less impact in this regard than the taller task links.
Still, a long queue will go offscreen anyway.

**Comment by timja** [comment]
Is it intended that the lines don't go all the way across for build executors?

image

**Review by timja — COMMENTED** [review]
Looks good apart from that line issue, which only appears to show in certain setups

**Comment by fqueiruga** [comment]
Setting https://github.com/jenkinsci/jenkins/blob/2980940aa8a0addec0aab0e41118076f1ef774e6/core/src/main/resources/lib/hudson/executors.jelly#L146  to 4 fixes it, but I'm not sure if it will break custom executor entries. Making the change anyway.

**Review by timja — APPROVED** [review]

**Review by uhafner — COMMENTED** [review]
The new styles look very good! (I did review the code)

**Review by uhafner — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
I am +1 for the change.  It indeed makes vertical layouts more difficult to read, but we should do something about build histories anyway (e.g. moving it to the main panel). It definitely improves the layout overall

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Comment by fqueiruga** [comment]
The problem there is that I cannot just add a padding to that entry without affecting other occurrences of that. I wonder if there's a way to just add the padding to the executors.

**Comment by timja** [comment]
Is it used other than in the side panel? Wouldn’t everything with collapsed text have the same issue?

**Comment by fqueiruga** [comment]
Thing is, I'm not sure. I don't know what else other plugins are doing there.

**Pull request merged**